avm/ptn/azd/aks/main.bicep

```diff
@@ -124,6 +122,66 @@ param containerRegistryRoleName string?
 @description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
 param aksClusterRoleAssignmentName string?
 
+@description('Optional. Custom configuration of system node pool.')
+param systemPoolConfig object = {}
+
+@description('Optional. Custom configuration of user node pool.')
+param agentPoolConfig object = {}
```

Comment:
You could consider adding 
`import {agentPoolType } from 'br/public:avm/res/container-service/managed-cluster:0.4.1'` in the file (note: would need to go on top of a @description block, not below) to re-use the type from the AKS module - if it satisfies your needs.

Reply:
Would look like this:
```bicep
import {agentPoolType } from 'br/public:avm/res/container-service/managed-cluster:0.4.1'
@description('Optional. Custom configuration of system node pool.')
param systemPoolConfig agentPoolType[]?

@description('Optional. Custom configuration of user node pool.')
param agentPoolConfig agentPoolType[]?
```
IF you want to use it

Reply:
Thanks for the suggestion, updated based on your comment.

Reply:
It's not exactly what I meant as you did not actually introduce a UDT which means the user needs to guess how the object must look like, but maybe something to implement at a later stage

Reply:
@AlexanderSehr Updated the code again based on your comment,